processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -608,7 +619,7 @@ private Assignment forgeMapping(SourceRHS sourceRHS) {
             Type returnType;
             // there's only one case for forging a method with mapping options: nested target properties.
             // they should always forge an update method
-            if ( method.isUpdateMethod() || forgeMethodWithMappingOptions != null ) {
+            if ( method.isUpdateMethod() || forceUpdateMethod ) {
```

Comment:
so if i understand you correctly you now have only update methods when you have nested target parameters and multiple source parameters, right? Is there a test case in which you check this?